lib/test_prof/factory_default.rb

```diff
@@ -9,18 +9,21 @@ module TestProf
   module FactoryDefault
     module DefaultSyntax # :nodoc:
       def create_default(name, *args, &block)
-        set_factory_default(
-          name,
-          TestProf::FactoryBot.create(name, *args, &block)
-        )
+        options = args.extract_options!
+        preserve = options.delete(:preserve_traits) || FactoryDefault.preserve_traits
+
+        obj = TestProf::FactoryBot.create(name, *args, options, &block)
+        set_factory_default(name, obj, preserve)
       end
 
-      def set_factory_default(name, obj)
-        FactoryDefault.register(name, obj)
+      def set_factory_default(name, obj, preserve_traits = nil)
```

Comment:
Let's make it a keyword argument and pass it as `.register(name, obj, options)`. We'll probably have more options in the future.